### ISSUE TITLE
Unable to install durable_rules 

### ISSUE THREAD

**Issue by nafey** [jruizgit/rules#69]
I have been trying to use durable_rules but the install is failing for me. 
I am currently using Ubuntu 14.04 with Python 2.7.6 and Redis 3.2.8 (followed the steps for installation from [here](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-redis))  

DETAILS
------------
According to Step 3 of the install steps of First App I try to execute `sudo pip install durable_rules` however I get the following error:

```
nafey@butler:~/_dev/rtest$ sudo pip install durable_rules
Downloading/unpacking durable-rules
  Downloading durable_rules-0.33.94.tar.gz (178kB): 178kB downloaded
  Running setup.py (path:/tmp/pip_build_root/durable-rules/setup.py) egg_info for package durable-rules
    
Downloading/unpacking werkzeug (from durable-rules)
  Downloading Werkzeug-0.12.1-py2.py3-none-any.whl (312kB): 312kB downloaded
Installing collected packages: durable-rules, werkzeug
  Running setup.py install for durable-rules
    building 'rules_py' library
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c deps/hiredis/hiredis.c -o build/temp.linux-x86_64-2.7/deps/hiredis/hiredis.o
    In file included from deps/hiredis/hiredis.c:42:0:
    deps/hiredis/hiredis.c: In function ‘__redisSetError’:
    deps/hiredis/hiredis.h:90:19: warning: ignoring return value of ‘strerror_r’, declared with attribute warn_unused_result [-Wunused-result]
             strerror_r((errno), (buf), (len));                                     \
                       ^
    deps/hiredis/hiredis.c:584:9: note: in expansion of macro ‘__redis_strerror_r’
             __redis_strerror_r(errno, c->errstr, sizeof(c->errstr));
             ^
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c deps/hiredis/net.c -o build/temp.linux-x86_64-2.7/deps/hiredis/net.o
    deps/hiredis/net.c: In function ‘__redisSetErrorFromErrno’:
    deps/hiredis/net.c:69:12: warning: variable ‘len’ set but not used [-Wunused-but-set-variable]
         size_t len = 0;
                ^
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c deps/hiredis/sds.c -o build/temp.linux-x86_64-2.7/deps/hiredis/sds.o
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c deps/hiredis/read.c -o build/temp.linux-x86_64-2.7/deps/hiredis/read.o
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c src/rules/json.c -o build/temp.linux-x86_64-2.7/src/rules/json.o
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c src/rules/net.c -o build/temp.linux-x86_64-2.7/src/rules/net.o
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c src/rules/rete.c -o build/temp.linux-x86_64-2.7/src/rules/rete.o
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c src/rules/state.c -o build/temp.linux-x86_64-2.7/src/rules/state.o
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c src/rules/events.c -o build/temp.linux-x86_64-2.7/src/rules/events.o
    src/rules/events.c: In function ‘assertTimers’:
    src/rules/events.c:1460:22: warning: ‘action’ may be used uninitialized in this function [-Wmaybe-uninitialized]
             unsigned int action;
                          ^
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c src/rules/regex.c -o build/temp.linux-x86_64-2.7/src/rules/regex.o
    ar rc build/temp.linux-x86_64-2.7/librules_py.a build/temp.linux-x86_64-2.7/deps/hiredis/hiredis.o build/temp.linux-x86_64-2.7/deps/hiredis/net.o build/temp.linux-x86_64-2.7/deps/hiredis/sds.o build/temp.linux-x86_64-2.7/deps/hiredis/read.o build/temp.linux-x86_64-2.7/src/rules/json.o build/temp.linux-x86_64-2.7/src/rules/net.o build/temp.linux-x86_64-2.7/src/rules/rete.o build/temp.linux-x86_64-2.7/src/rules/state.o build/temp.linux-x86_64-2.7/src/rules/events.o build/temp.linux-x86_64-2.7/src/rules/regex.o
    building 'rules' extension
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -Isrc/rules -I/usr/include/python2.7 -c src/rulespy/rules.c -o build/temp.linux-x86_64-2.7/src/rulespy/rules.o
    src/rulespy/rules.c:1:20: fatal error: Python.h: No such file or directory
     #include <Python.h>
                        ^
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    Complete output from command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip_build_root/durable-rules/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-wsq0q8-record/install-record.txt --single-version-externally-managed --compile:
    running install

running build

running build_py

creating build

creating build/lib.linux-x86_64-2.7

creating build/lib.linux-x86_64-2.7/durable

copying libpy/durable/__init__.py -> build/lib.linux-x86_64-2.7/durable

copying libpy/durable/lang.py -> build/lib.linux-x86_64-2.7/durable

copying libpy/durable/engine.py -> build/lib.linux-x86_64-2.7/durable

copying libpy/durable/interface.py -> build/lib.linux-x86_64-2.7/durable

running build_clib

building 'rules_py' library

creating build/temp.linux-x86_64-2.7

creating build/temp.linux-x86_64-2.7/deps

creating build/temp.linux-x86_64-2.7/deps/hiredis

creating build/temp.linux-x86_64-2.7/src

creating build/temp.linux-x86_64-2.7/src/rules

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c deps/hiredis/hiredis.c -o build/temp.linux-x86_64-2.7/deps/hiredis/hiredis.o

In file included from deps/hiredis/hiredis.c:42:0:

deps/hiredis/hiredis.c: In function ‘__redisSetError’:

deps/hiredis/hiredis.h:90:19: warning: ignoring return value of ‘strerror_r’, declared with attribute warn_unused_result [-Wunused-result]

         strerror_r((errno), (buf), (len));                                     \

                   ^

deps/hiredis/hiredis.c:584:9: note: in expansion of macro ‘__redis_strerror_r’

         __redis_strerror_r(errno, c->errstr, sizeof(c->errstr));

         ^

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c deps/hiredis/net.c -o build/temp.linux-x86_64-2.7/deps/hiredis/net.o

deps/hiredis/net.c: In function ‘__redisSetErrorFromErrno’:

deps/hiredis/net.c:69:12: warning: variable ‘len’ set but not used [-Wunused-but-set-variable]

     size_t len = 0;

            ^

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c deps/hiredis/sds.c -o build/temp.linux-x86_64-2.7/deps/hiredis/sds.o

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c deps/hiredis/read.c -o build/temp.linux-x86_64-2.7/deps/hiredis/read.o

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c src/rules/json.c -o build/temp.linux-x86_64-2.7/src/rules/json.o

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c src/rules/net.c -o build/temp.linux-x86_64-2.7/src/rules/net.o

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c src/rules/rete.c -o build/temp.linux-x86_64-2.7/src/rules/rete.o

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c src/rules/state.c -o build/temp.linux-x86_64-2.7/src/rules/state.o

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c src/rules/events.c -o build/temp.linux-x86_64-2.7/src/rules/events.o

src/rules/events.c: In function ‘assertTimers’:

src/rules/events.c:1460:22: warning: ‘action’ may be used uninitialized in this function [-Wmaybe-uninitialized]

         unsigned int action;

                      ^

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -c src/rules/regex.c -o build/temp.linux-x86_64-2.7/src/rules/regex.o

ar rc build/temp.linux-x86_64-2.7/librules_py.a build/temp.linux-x86_64-2.7/deps/hiredis/hiredis.o build/temp.linux-x86_64-2.7/deps/hiredis/net.o build/temp.linux-x86_64-2.7/deps/hiredis/sds.o build/temp.linux-x86_64-2.7/deps/hiredis/read.o build/temp.linux-x86_64-2.7/src/rules/json.o build/temp.linux-x86_64-2.7/src/rules/net.o build/temp.linux-x86_64-2.7/src/rules/rete.o build/temp.linux-x86_64-2.7/src/rules/state.o build/temp.linux-x86_64-2.7/src/rules/events.o build/temp.linux-x86_64-2.7/src/rules/regex.o

running build_ext

building 'rules' extension

creating build/temp.linux-x86_64-2.7/src/rulespy

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -std=c99 -D_GNU_SOURCE -fPIC -Isrc/rules -I/usr/include/python2.7 -c src/rulespy/rules.c -o build/temp.linux-x86_64-2.7/src/rulespy/rules.o

src/rulespy/rules.c:1:20: fatal error: Python.h: No such file or directory

 #include <Python.h>

                    ^

compilation terminated.

error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

----------------------------------------
Cleaning up...
Command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip_build_root/durable-rules/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-wsq0q8-record/install-record.txt --single-version-externally-managed --compile failed with error code 1 in /tmp/pip_build_root/durable-rules
Traceback (most recent call last):
  File "/usr/bin/pip", line 9, in <module>
    load_entry_point('pip==1.5.4', 'console_scripts', 'pip')()
  File "/usr/lib/python2.7/dist-packages/pip/__init__.py", line 235, in main
    return command.main(cmd_args)
  File "/usr/lib/python2.7/dist-packages/pip/basecommand.py", line 161, in main
    text = '\n'.join(complete_log)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 40: ordinal not in range(128)
nafey@butler:~/_dev/rtest$ python -V
Python 2.7.6
nafey@butler:~/_dev/rtest$ redis
No command 'redis' found, did you mean:
 Command 'redir' from package 'redir' (universe)
redis: command not found
nafey@butler:~/_dev/rtest$ redis-cli -v
redis-cli 3.2.8 (git:a346a62c)
```

Any help will be greatly appreciated

**Comment by nafey** [comment]
I am sorry this issue went away after I did a  `sudo apt-get install python-dev`

Closing this issue

**Comment by jruizgit** [comment]
Cool, thanks a lot for reporting this.

Sent from my iPhone

On Mar 28, 2017, at 2:38 AM, ynafey <notifications@github.com<mailto:notifications@github.com>> wrote:


Closed #69<https://github.com/jruizgit/rules/issues/69>.
